webserver/webserver/src/main/java/io/helidon/webserver/ByteBufDataChunk.java

```diff
@@ -0,0 +1,180 @@
+     * @param readOnly marks this buffer as read only
+     * @param releaseCallback a callback which is called when this chunk is completely processed
+     * @param byteBufs the data for this chunk. Should not be reused until {@code releaseCallback} is used
+     * @return new chunk
+     */
+    public static ByteBufDataChunk create(boolean flush, boolean readOnly, Runnable releaseCallback, ByteBuf... byteBufs) {
+        return new ByteBufDataChunk(flush, readOnly, releaseCallback, byteBufs);
+    }
+
+    /**
+     * Creates an instance given an array of {@code ByteBuf}'s.
+     *
+     * @param flush a signal that this chunk should be written and flushed from any cache if possible
+     * @param readOnly marks this buffer as read only
+     * @param byteBufs the data for this chunk. Should not be reused until {@code releaseCallback} is used
+     */
+    private ByteBufDataChunk(boolean flush, boolean readOnly, ByteBuf... byteBufs) {
+        this(flush, readOnly, null, byteBufs);
+    }
+
+    /**
+     * Creates an instance given an array of {@code ByteBuf}'s.
+     *
+     * @param flush a signal that this chunk should be written and flushed from any cache if possible
+     * @param readOnly marks this buffer as read only
+     * @param releaseCallback a callback which is called when this chunk is completely processed
+     * @param byteBufs the data for this chunk. Should not be reused until {@code releaseCallback} is used
+     */
+    private ByteBufDataChunk(boolean flush, boolean readOnly, Runnable releaseCallback, ByteBuf... byteBufs) {
+        this.flush = flush;
+        this.readOnly = readOnly;
+        this.releaseCallback = releaseCallback;
+        this.byteBufs = Objects.requireNonNull(byteBufs, "byteBuffers is null");
+    }
+
+    @Override
+    public <T> boolean isBackedBy(Class<T> clazz) {
+        return ByteBuf.class.isAssignableFrom(clazz);
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public <T> T[] data(Class<T> clazz) {
+        return (T[]) byteBufs;
+    }
+
+    @Override
+    public boolean isReleased() {
+        return isReleased;
+    }
+
+    @Override
+    public boolean flush() {
+        return flush;
+    }
+
+    @Override
+    public boolean isReadOnly() {
+        return readOnly;
+    }
+
+    @Override
+    public void release() {
+        if (!isReleased) {
+            if (releaseCallback != null) {
+                releaseCallback.run();
+            }
+            isReleased = true;
+        }
+    }
+
+    @Override
+    public void writeFuture(CompletableFuture<DataChunk> writeFuture) {
+        this.writeFuture = writeFuture;
+    }
+
+    @Override
+    public Optional<CompletableFuture<DataChunk>> writeFuture() {
+        return Optional.ofNullable(writeFuture);
+    }
+
+    @Override
+    public int remaining() {
+        int remaining = 0;
+        for (ByteBuf byteBuf : byteBufs) {
```

Comment:
If you're going to do stuff with multiple `ByteBuf` instances, consider Netty's built-in `CompositeByteBuf`.  It may or may not suit your needs.